src/UCTSearch.cpp

```diff
@@ -638,7 +641,7 @@ void UCTSearch::set_playout_limit(int playouts) {
                                       decltype(m_maxplayouts)>::value,
                   "Inconsistent types for playout amount.");
     if (playouts == 0) {
-        m_maxplayouts = std::numeric_limits<decltype(m_maxplayouts)>::max();
+        m_maxplayouts = std::numeric_limits<decltype(m_maxplayouts)>::max()/2;
```

Comment:
This is a bit sad and hacky, but I don't see a cleaner solution. Doing the m_maxplayouts check in the same place as the increment is not very clean either, and still requiring making the code robust against the overflow.

Maybe add a comment about avoiding overflow.

Reply:
Another option might be to not increment m_playouts if it is already at the m_maxplayouts in UCTSearch::increment_playouts().
Both variables could be changed to unsigned if there really is a desire to not limit them to half.

Reply:
Also just noticed on line 600 there is `(m_playouts * 100)` which could still overflow. I suppose that overflow is not critical but it might cause some confusion.

Reply:
See below. 1) improved code. 2) test proposed change. 3) test that the old way would have broken. I guess it probably isn't necessary to worry about it still overflowing when elapsed_centis is less than 100.
```
        // Calculate in two steps to lower chances of overflow.
        auto nps = m_playouts / (elapsed_centis+1);
        nps *= 100;
        myprintf("%d visits, %d nodes, %d playouts, %d n/s\n\n",
                 m_root->get_visits(),
                 static_cast<int>(m_nodes),
                 static_cast<int>(m_playouts),
                 nps);
        nps = std::numeric_limits<decltype(m_maxvisits)>::max()/2 / (elapsed_centis+1);
        nps *= 100;
        myprintf("debug1 %d visits, %d nodes, %d playouts, %d n/s\n\n",
                 m_root->get_visits(),
                 static_cast<int>(m_nodes),
                 static_cast<int>(m_playouts),
                 nps);
        nps = (std::numeric_limits<decltype(m_maxvisits)>::max() * 100) / (elapsed_centis+1);
        myprintf("debug2 %d visits, %d nodes, %d playouts, %d n/s\n\n",
                 m_root->get_visits(),
                 static_cast<int>(m_nodes),
                 static_cast<int>(m_playouts),
                 nps);
```

Example output, note 2nd works and 3rd is broken.
```
13 visits, 4673 nodes, 12 playouts, 200 n/s

debug1 13 visits, 4673 nodes, 12 playouts, 715827816 n/s

debug2 13 visits, 4673 nodes, 12 playouts, -16 n/s
```

Reply:
Oh this is easier. Also the other way was rounding to nearest 100 n/s.
```
        myprintf("%d visits, %d nodes, %d playouts, %.0f n/s\n\n",
                 m_root->get_visits(),
                 static_cast<int>(m_nodes),
                 static_cast<int>(m_playouts),
                 (m_playouts * 100.0) / (elapsed_centis+1));
```